kafka_consumer_freshness_tracker/src/test/java/com/tesla/data/consumer/freshness/ConsumerFreshnessTest.java

```diff
@@ -272,6 +275,127 @@ public void testBurrowMissingConsumerGroupPartitionEndOffsetMarksMissing() throw
     });
   }
 
+  @Test
+  public void testConfigurationContainsUnknownCluster() throws Exception {
+    Burrow burrow = mock(Burrow.class);
+    String clusterName = "bad_cluster";
+    when(burrow.getClusterBootstrapServers(clusterName)).thenThrow(new IOException());
+    Map<String, Object> conf = mockConfForCluster(
+            clusterName, "l1.example.com:9092", "l2.example.com:9092"
+    );
+
+    ConsumerFreshness freshness = new ConsumerFreshness();
+    freshness.burrow = burrow;
+
+    Optional<String> msg = freshness.validateClusterConf(conf);
+    Assert.assertTrue(msg.isPresent());
+    Assert.assertTrue(msg.get().contains("failed to read cluster detail from Burrow: "));
+    Assert.assertEquals(1.0,
+            freshness.getMetricsForTesting().burrowClusterDetailReadFailed.labels(clusterName).get(), 0.0);
+  }
+
+  @Test
+  public void testNormalModeConfigurationIsValidServersMatchBurrow() throws Exception {
+    Burrow burrow = mock(Burrow.class);
+    String clusterName = "cluster1";
+    when(burrow.getClusterBootstrapServers(clusterName))
+            .thenReturn(Arrays.asList("kafka01.example.com:10251", "kafka02.example.com:10251")
+      );
+
+    Map<String, Object> conf = mockConfForCluster(
+        clusterName,
+        "kafka01.example.com:10251", "kafka02.example.com:10251"
+        );
+
+    ConsumerFreshness freshness = new ConsumerFreshness();
+    freshness.burrow = burrow;
+
+    Assert.assertEquals(Optional.empty(), freshness.validateClusterConf(conf));
+  }
+
+  @Test
+  public void testNormalModeConfigurationIsInvalidContainsUnknownServer() throws Exception {
+    Burrow burrow = mock(Burrow.class);
+    String clusterName = "cluster1";
+    when(burrow.getClusterBootstrapServers(clusterName))
+      .thenReturn(Arrays.asList(
+            "kafka01.example.com:10251", "kafka02.example.com:10251")
+          );
+
+    Map<String, Object> conf = mockConfForCluster(
+        clusterName,
+        "kafka01.example.com:10251", "kafka02.example.com:10251", "kafka03.example.com:10251"
+        );
+
+    ConsumerFreshness freshness = new ConsumerFreshness();
+    freshness.burrow = burrow;
+
+    String expected = "bootstrap.servers contains the following servers which Burrow doesn't advertise: " +
+            "kafka03.example.com:10251";
+
+    Assert.assertEquals(Optional.of(expected), freshness.validateClusterConf(conf));
+  }
+
+  @Test
+  public void testStrictModeConfigurationIsValidServersMatchBurrow() throws Exception {
```

Comment:
you could just roll this into the first test method and show that matching servers don't have an error regardless of mode